### ISSUE TITLE
[BUG] no-magic-number: allowed cast error

### ISSUE THREAD

**Issue by furaiev** [dart-code-checker/dart-code-metrics#277]
* **Dart code metrics version:** 3.0.0
* **Dart sdk version:**  2.12.2 (stable)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
dart_code_metrics:
  anti-patterns:
    - long-method
    - long-parameter-list
  metrics:
    cyclomatic-complexity: 22
    lines-of-executable-code: 70
    number-of-parameters: 4
    maximum-nesting-level: 5
  metrics-exclude:
    - test/**
  rules:
    - newline-before-return
    - no-boolean-literal-compare
    - no-empty-block
    - prefer-trailing-comma
    - no-equal-then-else
    - avoid-unused-parameters
    - binary-expression-operand-order
    - double-literal-format
    - no-magic-number:
        allowed: [0.5, 1, 2, 3, 4, 5, 10, 100]
    - no-object-declaration
    - prefer-intl-name
    - provide-correct-intl-args
```
</details>

**What did you do? Please include the source code example causing the issue.**
run `flutter pub run dart_code_metrics:metrics lib`

**What did you expect to happen?**
Get a report

**What actually happened?**
```
Unhandled exception:
type 'List<Object>' is not a subtype of type 'Iterable<num>?' in type cast
```

<details>
<summary>Stack trace</summary>

```
Unhandled exception:
type 'List<Object>' is not a subtype of type 'Iterable<num>?' in type cast
#0      NoMagicNumberRule._parseConfig (package:dart_code_metrics/src/obsoleted/rules/no_magic_number_rule.dart:80:25)
#1      new NoMagicNumberRule (package:dart_code_metrics/src/obsoleted/rules/no_magic_number_rule.dart:21:32)
#2      _implementedRules.<anonymous closure> (package:dart_code_metrics/src/obsoleted/rules_factory.dart:43:41)
#3      getRulesById.<anonymous closure> (package:dart_code_metrics/src/obsoleted/rules_factory.dart:63:35)
#4      MappedIterator.moveNext (dart:_internal/iterable.dart:389:20)
#5      new _GrowableList._ofOther (dart:core-patch/growable_array.dart:198:26)
#6      new _List._ofOther (dart:core-patch/array.dart:106:43)
#7      new _List.of (dart:core-patch/array.dart:57:18)
#8      new List.of (dart:core-patch/array_patch.dart:52:20)
#9      new List.from (dart:core-patch/array_patch.dart:36:19)
#10     new List.unmodifiable (dart:core-patch/array_patch.dart:70:24)
#11     getRulesById (package:dart_code_metrics/src/obsoleted/rules_factory.dart:60:10)
#12     new MetricsAnalyzer (package:dart_code_metrics/src/obsoleted/metrics_analyzer.dart:53:22)
#13     _runAnalysis (file:///Users/admin/developer/flutter/.pub-cache/hosted/pub.dartlang.org/dart_code_metrics-3.0.0/bin/metrics.dart:47:20)
<asynchronous suspension>
#14     main (file:///Users/admin/developer/flutter/.pub-cache/hosted/pub.dartlang.org/dart_code_metrics-3.0.0/bin/metrics.dart:26:5)
<asynchronous suspension>
pub finished with exit code 255

```

</details>
